Readability.js

```diff
@@ -155,7 +155,15 @@ Readability.prototype = {
    * @return void
    */
   _forEachNode: function(nodeList, fn) {
-    return Array.prototype.forEach.call(nodeList, fn, this);
+    var i = 0;
+    while (nodeList[i]) {
+      var el = nodeList[i];
+      fn.call(this, el, i, nodeList);
+      // Only increment i if the element is still in the list:
+      if (nodeList[i] == el) {
+        i++
+      }
+    }
```

Comment:
The only thing that I don't like here is fact that if you delete any node, next node will have the same index as previous. It doesn't match to the original behaviour of `forEach` and may be unobvious for users in the future.